packages/react-storage/src/components/StorageBrowser/views/LocationActionView/UploadControls.tsx

```diff
@@ -294,14 +295,20 @@ export const UploadControls = (): JSX.Element => {
       taskCounts.TOTAL;
 
   const disableCancel = !taskCounts.TOTAL || !hasStarted || hasCompleted;
-  const disablePrimary = !taskCounts.TOTAL || hasStarted || hasCompleted;
```

Comment:
This is the kind of disable logic I expect to exist outside of the control given that the logic could vary from view to view